org.lflang/src/org/lflang/generator/ts/TSConstructorGenerator.kt

```diff
@@ -67,61 +66,62 @@ class TSConstructorGenerator (
         return arguments.joinToString(", \n")
     }
 
-    private fun federationRTIProperties(): LinkedHashMap<String, Any> {
-        return tsGenerator.federationRTIPropertiesW()
-    }
-
-    private fun generateSuperConstructorCall(reactor: Reactor, federate: FederateInstance): String {
+    private fun generateSuperConstructorCall(reactor: Reactor): String {
         if (reactor.isMain) {
             return "super(timeout, keepAlive, fast, success, fail);"
-        } else if (reactor.isFederated) {
-            var port = federationRTIProperties()["port"]
-            // Default of 0 is an indicator to use the default port, 15045.
-            if (port == 0) {
-                port = 15045
-            }
-            return """
-            super(federationID, ${federate.id}, ${port},
-                "${federationRTIProperties()["host"]}",
-                timeout, keepAlive, fast, success, fail);
-            """
-        } else {
+        }
+        // FIXME: Move this to FedGenerator
+//        else if (reactor.isFederated) {
+//            var port = federationRTIProperties()["port"]
+//            // Default of 0 is an indicator to use the default port, 15045.
+//            if (port == 0) {
+//                port = 15045
+//            }
+//            return """
+//            super(federationID, ${federate.id}, ${port},
+//                "${federationRTIProperties()["host"]}",
+//                timeout, keepAlive, fast, success, fail);
+//            """
+//        }
```

Comment:
@lhstrh, there's an issue when generating the TypeScript code here.
Currently, `main reactor` is generated as `App` and `federated reactor` is generated as `FederatedApp`. After fed-gen, all the reactors will become `main reactor`, and will be generated as `App`. Thus, we won't be able to use federated specific functions in this case. Do you have any suggestions on how to address this issue?

@Soroosh129, is there a way to distinguish whether a .lf file was generated by `fed-gen` or manually written by user?

Reply:
> @Soroosh129, is there a way to distinguish whether a .lf file was generated by fed-gen or manually written by user?

Not currently, but we can easily add that if needed (it could be as simple as a special marker in the preamble).

Reply:
> Not currently, but we can easily add that if needed (it could be as simple as a special marker inside the preamble)
Thank you, @Soroosh129 ! I think this will be very helpful. Maybe we need some way to indicate the LF is meant to be converted to a federated reactor anyway. Do you think you could add that to this PR?

Reply:
> Do you think you could add that to this PR?

Do you mean we should add this for all targets? Why not just add a special marker for the TS target's preamble in TSExtension?

Reply:
> Why not just add a special marker for the TS target's preamble in TSExtension?

It's because we should have this information at the code generation time in the current design. Federated and main reactors are code-generated differently like these (they inherit different classes):

- `class HelloDistributed extends __FederatedApp`
- `class HelloDistributed extends __App`

> Do you mean we should add this for all targets?

I think this will be useful, what do you think?

Reply:
> I think this will be useful, what do you think?

We don't have a use case for that in the C and Python targets , at least for now.

> It's because we should have this information at the code generation time in the current design.

Right. The special marker is meant to be read by the code generator. For example:
```TypeScript
preamble {=
    // 0f6cf1 This program is federated
=}
```